Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -85,164 +116,256 @@ FieldProbe::FieldProbe (std::string rd_name)
+
     const auto nLevel = warpx.finestLevel() + 1;
 
-    // vector where we store the field values
+    // vector to store the field values
+
     amrex::Vector<amrex::Real> fp_values(noutputs, 0);
 
-    // loop over refinement levels
+   	// loop over refinement levels
+
     for (int lev = 0; lev < nLevel; ++lev)
     {
-
         const amrex::Geometry& gm = warpx.Geom(lev);
         const auto prob_lo = gm.ProbLo();
         const auto prob_hi = gm.ProbHi();
 
-#if (AMREX_SPACEDIM == 2)
-        m_probe_in_domain = x_probe >= prob_lo[0] and x_probe < prob_hi[0] and
-                            z_probe >= prob_lo[1] and z_probe < prob_hi[1];
-#else
-        m_probe_in_domain = x_probe >= prob_lo[0] and x_probe < prob_hi[0] and
-                            y_probe >= prob_lo[1] and y_probe < prob_hi[1] and
-                            z_probe >= prob_lo[2] and z_probe < prob_hi[2];
-#endif
-        if(lev == 0) m_probe_in_domain_lev_0 = m_probe_in_domain;
-
-        if( !m_probe_in_domain ) break;
-
-        const auto cell_size = gm.CellSizeArray();
-
-        const int i_probe = static_cast<int>(amrex::Math::floor((x_probe - prob_lo[0]) / cell_size[0]));
-#if (AMREX_SPACEDIM == 2)
-        const int j_probe = static_cast<int>(amrex::Math::floor((z_probe - prob_lo[1]) / cell_size[1]));
-        const int k_probe = 0;
-#elif(AMREX_SPACEDIM == 3)
-        const int j_probe = static_cast<int>(amrex::Math::floor((y_probe - prob_lo[1]) / cell_size[1]));
-        const int k_probe = static_cast<int>(amrex::Math::floor((z_probe - prob_lo[2]) / cell_size[2]));
-#endif
         // get MultiFab data at lev
+
         const amrex::MultiFab &Ex = warpx.getEfield(lev, 0);
         const amrex::MultiFab &Ey = warpx.getEfield(lev, 1);
         const amrex::MultiFab &Ez = warpx.getEfield(lev, 2);
         const amrex::MultiFab &Bx = warpx.getBfield(lev, 0);
         const amrex::MultiFab &By = warpx.getBfield(lev, 1);
         const amrex::MultiFab &Bz = warpx.getBfield(lev, 2);
 
-        // Prepare interpolation of field components to probe_position
-        // The arrays below store the index type (staggering) of each MultiFab.
+        /*
+         * Prepare interpolation of field components to probe_position
+         * The arrays below store the index type (staggering) of each MultiFab.
+         */
+
         amrex::IndexType const Extype = Ex.ixType();
         amrex::IndexType const Eytype = Ey.ixType();
         amrex::IndexType const Eztype = Ez.ixType();
         amrex::IndexType const Bxtype = Bx.ixType();
         amrex::IndexType const Bytype = By.ixType();
         amrex::IndexType const Bztype = Bz.ixType();
 
+        //defined for use in determining which CPU contains the particles
+
         int probe_proc = -1;
-        // MFIter loop to interpolate fields to probe position
-#ifdef AMREX_USE_OMP
-#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+
+	    //loop over each particle
+
+    	using MyParIter = amrex::ParIter<0,0,7,0>;
```

Comment:
```suggestion
    	using MyParIter = amrex::ParIter<0,0,static_cast<int>(ParticleVal::nattribs),0>;
```

or even better: go into `FieldProbeParticleContainer` in the `public` section and do:
```C++
using iterator = amrex::ParIter<0,0,static_cast<int>(ParticleVal::nattribs),0>;
```
and then use here:
```
using MyParIter = FieldProbeParticleContainer::iterator;
```

Reply:
Implemented locally